### PR TITLE
Feat/code refactoring

### PR DESCRIPTION
**Pull request by FabianRuettgers**

### CODE REVIEW

File: components/Buttons/DeleteCommentButton/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            height="24"
+            viewBox="0 -960 960 960"
+            width="24"
+          >
+            <path d="M261-120q-24.75 0-42.375-17.625T201-180v-570h-41v-60h188v-30h264v30h188v60h-41v570q0 24-18 42t-42 18H261Zm438-630H261v570h438v-570ZM367-266h60v-399h-60v399Zm166 0h60v-399h-60v399ZM261-750v570-570Z" />
+          </svg>
+        </StyledButton>
+      ) : null}
```
Comment (Roland-Hufnagel) [inline components/Buttons/DeleteCommentButton/index.js]: You should not render a button and give him a prop to determine if he renders anything or not!
The ``editMode`` should be used, where the button is called (or not).

File: components/Buttons/EditCommentButton/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { styled } from "styled-components";
+
+export default function EditCommentButton({
+  handleEditClick,
+  handleBackClick,
+  editMode,
+}) {
+  console.log(editMode);
```
Comment (Roland-Hufnagel) [inline components/Buttons/EditCommentButton/index.js]: please delete

File: components/Buttons/EditCommentButton/index.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { styled } from "styled-components";
+
+export default function EditCommentButton({
```
Comment (Roland-Hufnagel) [inline components/Buttons/EditCommentButton/index.js]: If the ``EditCommentButton``can also conditionally render a ``Back``-Button, then the name `ÈditCommentButton``is a bit confusing.

File: components/Buttons/EditCommentButton/index.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  handleEditClick,
+  handleBackClick,
+  editMode,
+}) {
+  console.log(editMode);
+  return (
+    <>
+      {editMode ? (
```
Comment (Roland-Hufnagel) [inline components/Buttons/EditCommentButton/index.js]: And it's the same thing here: if there are two possibilities you should better decide it at the place where you call it.

File: components/Buttons/SaveCommentButton/index.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { styled } from "styled-components";
+
+export default function SaveCommentButton({ onClick, disabled, editMode }) {
+  return (
+    <>
+      {editMode ? (
```
Comment (Roland-Hufnagel) [inline components/Buttons/SaveCommentButton/index.js]: The same again.

File: components/MovieDetailPage/MovieCommentForm/CreateMovieComment/index.js [hunk 6]
```diff
@@ -7,34 +7,45 @@ export default function CreateMovieComment({
...
-          <FormButton
-            title={"zurück"}
-            backgroundcolor={"var(--background-color-light)"}
-            textcolor={"var( --text-color-dark-content)"}
-            handleClick={handleCommentButtonClick}
+    <>
+      <StyledButton onClick={handleCommentButtonClick} />
+      <Container>
+        <Heading>Kommentar</Heading>
+        <Form onSubmit={handleComment(id)}>
```
Comment (Roland-Hufnagel) [inline components/MovieDetailPage/MovieCommentForm/CreateMovieComment/index.js:14]: ``handleComment(id)`` is a function-call. This cannot work.

File: components/MovieDetailPage/MovieDetails/MovieComments/index.js [hunk 7]
```diff
@@ -1,33 +1,47 @@
...
+  const emptyInput =
+    !currentAction.editingComment || !currentAction.editingComment.content;
   const foundUser = userInformation.find((user) => user.id === id);
   const userComments = foundUser ? foundUser.comments : null;
   const comments = MovieComments.results;
   const combinedComments = userComments
     ? [...userComments, ...comments]
     : comments;
 
+  function deleteMode(commentId) {
```
Comment (Roland-Hufnagel) [inline components/MovieDetailPage/MovieDetails/MovieComments/index.js]: A function name should describe what the function does. Is this function deleting a Mode?

File: components/MovieDetailPage/MovieDetails/MovieComments/index.js [hunk 8]
```diff
@@ -1,33 +1,47 @@
...
+    const itemToDelete =
+      currentAction.userInput === "ACTION_DELETE_COMMENT" &&
+      currentAction.editingComment &&
+      currentAction.editingComment.id === commentId;
+    return itemToDelete;
+  }
+
+  function editMode(commentId) {
```
Comment (Roland-Hufnagel) [inline components/MovieDetailPage/MovieDetails/MovieComments/index.js]: same as above

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]
Hi. There is a 'MovieDetailPage' within your component folder. That makes me confused. What is it? If it is not a page please don't call it so. 
And in the preview I saw some unexpected behaviours when adding comments and scrolling. There are some bugs in your stacking context. 
I left some comments. Look at them and try to improve your code.

**Comment by Faibs7** [comment]
I find your changes in naming convention, which Roland addressed on several points, much more understandable and better now. Btw i find your Z-index with 999999999 awesome :D. Also the addressed points concerning the button were well implemented. Looks great and I have noticed nothing else. LGTM

**Pull request merged**